controller_manager/test/test_load_controller.cpp

```diff
@@ -337,240 +263,161 @@ INSTANTIATE_TEST_SUITE_P(
-    ControllerManagerRunner cm_runner(this);
-    EXPECT_EQ(controller_interface::return_type::OK, switch_future.get());
+TEST_F(TestTwoLoadedControllers, load_and_configure_two_known_controllers)
+{
+  cm_->configure_controller(controller_name1);
+  EXPECT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if1->get_state().id());
 
-    ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE, controller_if1->get_state().id());
-    ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if2->get_state().id());
-  }
+  cm_->configure_controller(controller_name2);
+  EXPECT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if2->get_state().id());
+}
 
-  {  // Stop controller 1, start controller 2
-    strvec start_controllers = {controller_name2};
-    strvec stop_controllers = {controller_name1};
-    RCLCPP_INFO(cm_->get_logger(), "Stopping controller #1, starting controller #2");
-    auto switch_future = std::async(
-      std::launch::async, &controller_manager::ControllerManager::switch_controller, cm_,
-      start_controllers, stop_controllers, STRICT, true, rclcpp::Duration(0, 0));
+TEST_F(TestTwoLoadedControllers, switch_multiple_controllers)
+{
+  // Only testing with STRICT now for simplicity
+  //  Test starting a stopped controller, and stopping afterwards
 
-    ASSERT_EQ(std::future_status::timeout, switch_future.wait_for(std::chrono::milliseconds(100)))
-      << "switch_controller should be blocking until next update cycle";
-    ControllerManagerRunner cm_runner(this);
-    EXPECT_EQ(controller_interface::return_type::OK, switch_future.get());
+  cm_->configure_controller(controller_name1);
 
-    ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if1->get_state().id());
-    ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE, controller_if2->get_state().id());
-  }
+  // Start controller #1
+  RCLCPP_INFO(cm_->get_logger(), "Starting stopped controller #1");
+  switch_test_controllers(strvec{controller_name1}, strvec{});
+  ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE, controller_if1->get_state().id());
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED, controller_if2->get_state().id());
 
-  {  // stop controller 2
-    strvec start_controllers = {};
-    strvec stop_controllers = {controller_name2};
-    RCLCPP_INFO(cm_->get_logger(), "Stopping controller #1, starting controller #2");
-    auto switch_future = std::async(
-      std::launch::async, &controller_manager::ControllerManager::switch_controller, cm_,
-      start_controllers, stop_controllers, STRICT, true, rclcpp::Duration(0, 0));
+  // Stop controller 1, start controller 2
+  // Both fail because controller 2 because it is not configured and STRICT is used
+  RCLCPP_INFO(
+    cm_->get_logger(),
+    "Stopping controller #1, starting unconfigured controller #2 fails (STRICT)");
```

Comment:
Could you also add the test with `BEST_EFFORT` for this? Probably completely new test-case is needed.